api/v1beta1/cryostat_conversion.go

```diff
@@ -0,0 +1,590 @@
+// Copyright The Cryostat Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package v1beta1
+
+import (
+	operatorv1beta2 "github.com/cryostatio/cryostat-operator/api/v1beta2"
+	"sigs.k8s.io/controller-runtime/pkg/conversion"
+)
+
+var _ conversion.Convertible = &Cryostat{}
+
+// ConvertTo converts this Cryostat to the Hub version (v1beta2).
+func (src *Cryostat) ConvertTo(dstRaw conversion.Hub) error {
+	dst := dstRaw.(*operatorv1beta2.Cryostat)
+
+	// Copy ObjectMeta as-is
+	dst.ObjectMeta = src.ObjectMeta
+
+	// Convert existing Spec fields
+	convertSpecTo(&src.Spec, &dst.Spec)
+
+	// Convert existing Status fields
+	convertStatusTo(&src.Status, &dst.Status)
+
+	// Maintain the previous behaviour by using the CR's namespace as the sole target namespace
+	dst.Spec.TargetNamespaces = []string{src.Namespace}
+	dst.Status.TargetNamespaces = []string{src.Namespace}
+
+	return nil
+}
+
+func convertSpecTo(src *CryostatSpec, dst *operatorv1beta2.CryostatSpec) {
+	dst.Minimal = src.Minimal
+	dst.EnableCertManager = src.EnableCertManager
+	dst.TrustedCertSecrets = convertCertSecretsTo(src.TrustedCertSecrets)
+	dst.EventTemplates = convertEventTemplatesTo(src.EventTemplates)
```

Comment:
3.0 hasn't yet reimplemented custom event templates, but when it does I'm expecting that to be done by using S3 storage rather than direct PVC access. That'll make it more complicated to support the feature from the CR level. Maybe we should drop this like we dropped the FlightRecording CRD?

Reply:
Would there still be a way to do this declaratively? I haven't really used this feature much, but I could see this being useful for GitOps-managed clusters.

Reply:
I guess there are two apparent routes:
1. the Operator uses an HTTP client and creates the custom event templates on the Cryostat application instances through the API like any other client
2. when we do reimplement custom event templates in 3.0, we do it by direct filesystem access (PVC) like we had before rather than S3. Or, use S3 as a read-write store in addition to a read-only direct filesystem PVC source where additional templates can be mounted.

Reply:
I think there are other cases of things that could be useful for things like GitOps clusters, like Automated Rules as a CRD. In 3.0 those are defined in a Postgres table so IMO the most sensible way would be for the Operator to talk to Cryostat via the HTTP API there, too.

Reply:
That seems reasonable. These sorts of features would likely never be possible for the Helm Chart, but that could be an advantage of using the operator instead.

Reply:
Then again, we do have other things that seem to make the most sense as files directly injected into the PVC still, like TLS certs...